### ISSUE TITLE
`cabal install hi` fails

### ISSUE THREAD

**Issue by nha** [fujimura/hi#61]
Quite new to Haskell, please correct me if I am doing it wrong (there was no "installation" on the README).

Here is the output, on OS X 10.11.6:

> $ cabal install hi
> Resolving dependencies...
> Configuring hi-1.2.0.1...
> Building hi-1.2.0.1...
> Failed to install hi-1.2.0.1
> Build log ( /Users/nicolas.ha/.cabal/logs/hi-1.2.0.1.log ):
> cabal: Entering directory '/var/folders/yr/v0tf__1d5pl2gs55bkqczcpw0000gr/T/cabal-tmp-10436/hi-1.2.0.1'
> Configuring hi-1.2.0.1...
> Building hi-1.2.0.1...
> Preprocessing library hi-1.2.0.1...
> [ 1 of 12] Compiling Hi.Config        ( src/Hi/Config.hs, dist/build/Hi/Config.o )
> [ 2 of 12] Compiling Hi.CommandLineOption ( src/Hi/CommandLineOption.hs, dist/build/Hi/CommandLineOption.o )
> 
> src/Hi/CommandLineOption.hs:23:39: error:
>     • Variable not in scope: (<>) :: Mod f0 a0 -> Mod f1 a1 -> t0
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:23:62: error:
>     • Variable not in scope:
>         (<>) :: t0 -> Mod f2 a2 -> Mod OptionFields String
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:26:39: error:
>     • Variable not in scope: (<>) :: Mod f3 a3 -> Mod f4 a4 -> t2
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:26:62: error:
>     • Variable not in scope: (<>) :: t2 -> Mod f5 a5 -> t1
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:26:85: error:
>     • Variable not in scope:
>         (<>) :: t1 -> Mod f6 a6 -> Mod OptionFields String
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:27:39: error:
>     • Variable not in scope: (<>) :: Mod f7 a7 -> Mod f8 a8 -> t3
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:27:64: error:
>     • Variable not in scope:
>         (<>) :: t3 -> Mod f9 a9 -> Mod OptionFields String
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:28:39: error:
>     • Variable not in scope: (<>) :: Mod f10 a10 -> Mod f11 a11 -> t4
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:28:62: error:
>     • Variable not in scope:
>         (<>) :: t4 -> Mod f12 a12 -> Mod OptionFields String
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:29:39: error:
>     • Variable not in scope: (<>) :: Mod f13 a13 -> Mod f14 a14 -> t5
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:29:62: error:
>     • Variable not in scope:
>         (<>) :: t5 -> Mod f15 a15 -> Mod OptionFields String
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:30:39: error:
>     • Variable not in scope: (<>) :: Mod f16 a16 -> Mod f17 a17 -> t7
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:30:62: error:
>     • Variable not in scope: (<>) :: t7 -> Mod f18 a18 -> t6
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:30:92: error:
>     • Variable not in scope:
>         (<>) :: t6 -> Mod f19 String -> Mod OptionFields String
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:31:62: error:
>     • Variable not in scope:
>         (<>) :: Mod f20 a19 -> Mod f21 a20 -> Mod OptionFields String
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:32:62: error:
>     • Variable not in scope:
>         (<>) :: Mod f22 a21 -> Mod f23 a22 -> Mod FlagFields Bool
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> 
> src/Hi/CommandLineOption.hs:33:62: error:
>     • Variable not in scope:
>         (<>) :: Mod f24 a23 -> Mod f25 a24 -> Mod OptionFields String
>     • Perhaps you meant one of these:
>         ‘<$>’ (imported from Options.Applicative),
>         ‘<*’ (imported from Options.Applicative),
>         ‘<|>’ (imported from Options.Applicative)
> cabal: Leaving directory '/var/folders/yr/v0tf__1d5pl2gs55bkqczcpw0000gr/T/cabal-tmp-10436/hi-1.2.0.1'
> cabal: Error: some packages failed to install:
> hi-1.2.0.1 failed during the building phase. The exception was:
> ExitFailure 1

**Comment by StefanYohansson** [comment]
Same here.

```
~ cabal install hi
Resolving dependencies...
Configuring hi-1.2.0.1...
Building hi-1.2.0.1...
Failed to install hi-1.2.0.1
Build log ( /home/snotr/.cabal/logs/hi-1.2.0.1.log ):
cabal: Entering directory '/tmp/cabal-tmp-18462/hi-1.2.0.1'
Configuring hi-1.2.0.1...
Building hi-1.2.0.1...
Preprocessing library hi-1.2.0.1...
[ 1 of 12] Compiling Hi.Config        ( src/Hi/Config.hs, dist/build/Hi/Config.o )
[ 2 of 12] Compiling Hi.CommandLineOption ( src/Hi/CommandLineOption.hs, dist/build/Hi/CommandLineOption.o )

src/Hi/CommandLineOption.hs:23:39: error:
    • Variable not in scope: (<>) :: Mod f0 a0 -> Mod f1 a1 -> t0
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:23:62: error:
    • Variable not in scope:
        (<>) :: t0 -> Mod f2 a2 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:26:39: error:
    • Variable not in scope: (<>) :: Mod f3 a3 -> Mod f4 a4 -> t2
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:26:62: error:
    • Variable not in scope: (<>) :: t2 -> Mod f5 a5 -> t1
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:26:85: error:
    • Variable not in scope:
        (<>) :: t1 -> Mod f6 a6 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:27:39: error:
    • Variable not in scope: (<>) :: Mod f7 a7 -> Mod f8 a8 -> t3
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:27:64: error:
    • Variable not in scope:
        (<>) :: t3 -> Mod f9 a9 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:28:39: error:
    • Variable not in scope: (<>) :: Mod f10 a10 -> Mod f11 a11 -> t4
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:28:62: error:
    • Variable not in scope:
        (<>) :: t4 -> Mod f12 a12 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:29:39: error:
    • Variable not in scope: (<>) :: Mod f13 a13 -> Mod f14 a14 -> t5
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:29:62: error:
    • Variable not in scope:
        (<>) :: t5 -> Mod f15 a15 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:30:39: error:
    • Variable not in scope: (<>) :: Mod f16 a16 -> Mod f17 a17 -> t7
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:30:62: error:
    • Variable not in scope: (<>) :: t7 -> Mod f18 a18 -> t6
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:30:92: error:
    • Variable not in scope:
        (<>) :: t6 -> Mod f19 String -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:31:62: error:
    • Variable not in scope:
        (<>) :: Mod f20 a19 -> Mod f21 a20 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:32:62: error:
    • Variable not in scope:
        (<>) :: Mod f22 a21 -> Mod f23 a22 -> Mod FlagFields Bool
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)

src/Hi/CommandLineOption.hs:33:62: error:
    • Variable not in scope:
        (<>) :: Mod f24 a23 -> Mod f25 a24 -> Mod OptionFields String
    • Perhaps you meant one of these:
        ‘<$>’ (imported from Options.Applicative),
        ‘<*’ (imported from Options.Applicative),
        ‘<|>’ (imported from Options.Applicative)
cabal: Leaving directory '/tmp/cabal-tmp-18462/hi-1.2.0.1'
cabal: Error: some packages failed to install:
hi-1.2.0.1 failed during the building phase. The exception was:
ExitFailure 1

~ ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.0.1

~ cabal --version
cabal-install version 1.24.0.0
compiled using version 1.24.0.0 of the Cabal library 
```

**Comment by StefanYohansson** [comment]
I have used https://github.com/trofi/hi repo:
- Clone it;
- cd into folder;
- `cabal run` to compile it;
- cd into your project folder;
- run `./whereyoucloned/hi/dist/build/hi/hi project-folder`

**Comment by fujimura** [comment]
Fixed in 1.2.0.2. see: https://github.com/fujimura/hi/issues/63#issuecomment-408402203

Thanks for reporting this!
